src/store.ts

```diff
@@ -5,11 +5,22 @@ import type { RootState, AppAction } from './reducers';
 
 type AppStore =
   Store<RootState, AppAction> &
-  // TODO(pcardune): this onKeyDown property gets set by
-  // BlockEditor and used by the Node component for no apparent
-  // reason. It's effectively a global variable and should be
-  // stored somewhere else.
-  {onKeyDown?: Function};
+  // TODO(pcardune): these additional properties are tacked onto the store
+  // in random places, but shouldn't be.
+  {
+    // TODO(pcardune): this onKeyDown property gets set by
+    // BlockEditor and used by the Node component for no apparent
+    // reason. It's effectively a global variable and should be
+    // stored somewhere else.
+    onKeyDown?: Function,
```

Comment:
@pcardune this is a dumb question, but: what's the canonical design pattern for where these go? Props on the entire editor? If so, how do they get passed down through all the children? I'm still a little iffy on the appropriate uses of Contexts, so maybe this is just my own blindspot...

Reply:
Without looking further into what the intended semantics are, I'd say that using a context is appropriate here. However, I don't yet understand what this keydown stuff is trying to accomplish at a high level, and depending on what bigger purpose its serving, it might be possible to do without global variables _or_ context. The `keyDown()` function in keymap.js has a lot of side effects so it's hard to understand what's going on.